aiorospy/src/aiorospy/helpers.py

```diff
@@ -88,7 +89,17 @@ def _task_done_callback(self, task):
                 self._exception_q.sync_q.put(exc)
 
 
-async def log_during(awaitable, msg, period, sink=logger.info):
+def iscoroutinefunction_or_partial(fxn):
+    """ Before python3.8, asyncio.iscoroutine is unable to examine coroutines wrapped via partial.
+    See https://stackoverflow.com/a/52422903/1198131
+    """
+    while isinstance(fxn, functools.partial):
+        fxn = fxn.func
+    return asyncio.iscoroutinefunction(fxn)
+
+
+async def do_while(awaitable, period, do, *args, **kwargs):
```

Comment:
FYI @garyservin, I've added this as a separate helper from log_during. I'll patch it into ota_service next time I'm in there.